Automattic/wp-calypso
# Add Do Not Sell privacy setting

#### Proposed Changes

- Add the Do Not Sell setting to the Privacy page (`/me/privacy/`)
  - Disable setting if cookie banner has not been yet accepted
  - Don't show the setting for non-CCPA regions 

#### Testing Instructions

* Checkout changes and run Calypso dev env
* Go to `calypso.localhost:3000/me/privacy?flags=cookie-banner`
* Set cookies for CCPA region: `country_code=US; region=California;`
* Set all buckets to accepted: `document.cookie = 'sensitive_pixel_options=%7B%22version%22%3A20201224%2C%22ok%22%3Atrue%2C%22buckets%22%3A%7B%22essential%22%3Atrue%2C%22analytics%22%3Atrue%2C%22advertising%22%3Atrue%7D%7D';`
* You should be able to see the Do Not Sell setting on Privacy page
<img width="1076" alt="CleanShot 2022-11-29 at 18 27 22@2x" src="https://user-images.githubusercontent.com/8419292/204678059-b99e1b81-9f39-4dd8-a0f3-030c18f1dbbe.png">

* You should be able to switch Do Not Sell setting - it should impact `buckets.advertising` value in `sensitive_pixel_options` cookie
* Remove the `sensitive_pixel_options` cookie (stay in CCPA region)
* You should see Do Not Sell setting on Privacy page, but the switch should be disabled
<img width="1067" alt="CleanShot 2022-11-29 at 18 26 23@2x" src="https://user-images.githubusercontent.com/8419292/204677959-b94fd781-3e47-4811-a540-a42348a2c64a.png">

* Change `country_code` to any other country e.g. `country_code=NL`
* The Do Not Sell setting should not appear on Privacy page
